File: string-paradigm-expression-original/src/main/java/com/steelzack/string/paradigm/original/expression/RegionSpellingCheck.java
package com.steelzack.string.paradigm.original.expression;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Pattern;

public class RegionSpellingCheck {

    private static final Pattern DOUBLE_COMMAS = Pattern.compile(".*,\\s*,.*");
    private static final Pattern WHITE_SPACES = Pattern.compile(".*\\s\\s+.*");

    private final List<RegionImpl> spellingDoubleCommaRegions = new ArrayList<RegionImpl>();
	private final List<RegionImpl> spellingWhiteSpacesRegions = new ArrayList<RegionImpl>();
	private final List<RegionImpl> spellingBeginCapitalRegions = new ArrayList<RegionImpl>();

    public void checkSpelling(RegionImpl component) {
    	if (DOUBLE_COMMAS.matcher(component.getDescription()).matches()) {
    	    spellingDoubleCommaRegions.add(component);
    	}
    	if (WHITE_SPACES.matcher(component.getDescription()).matches()) {
    	    spellingWhiteSpacesRegions.add(component);
    	}
    	String[] elements = component.getDescription().split(",");
    	for (int i = 0; i < elements.length; i++) {
    		String element = elements[i].trim();
    		if (element.length() > 0 && Character.getType(element.charAt(0)) == Character.LOWERCASE_LETTER) {
    		    spellingBeginCapitalRegions.add(component);
    		}
    	}
    }

	public List<RegionImpl> getSpellingDoubleCommaRegions() {
		return spellingDoubleCommaRegions;
	}

	public List<RegionImpl> getSpellingWhiteSpacesRegions() {
		return spellingWhiteSpacesRegions;
	}

	public List<RegionImpl> getSpellingBeginCapitalRegions() {
		return spellingBeginCapitalRegions;
	}
}

File: string-paradigm-expression-original/src/main/kotlin/org/jesperancinha/string/paradigm/original/expression/RegionSpellingCheck.kt
package org.jesperancinha.string.paradigm.original.expression

import java.util.regex.Pattern

class RegionSpellingCheck {
    val spellingDoubleCommaRegions: MutableList<RegionImpl> = ArrayList()
    val spellingWhiteSpacesRegions: MutableList<RegionImpl> = ArrayList()
    val spellingBeginCapitalRegions: MutableList<RegionImpl> = ArrayList()
    fun checkSpelling(component: RegionImpl) {
        if (DOUBLE_COMMAS.matcher(component.description).matches()) {
            spellingDoubleCommaRegions.add(component)
        }
        if (WHITE_SPACES.matcher(component.description).matches()) {
            spellingWhiteSpacesRegions.add(component)
        }
        val elements = component.description.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (i in elements.indices) {
            val element = elements[i].trim { it <= ' ' }
            if (element.length > 0 && Character.getType(element[0]) == Character.LOWERCASE_LETTER.toInt()) {
                spellingBeginCapitalRegions.add(component)
            }
        }
    }

    companion object {
        private val DOUBLE_COMMAS = Pattern.compile(".*,\\s*,.*")
        private val WHITE_SPACES = Pattern.compile(".*\\s\\s+.*")
    }
}